File: PHP/GincanaNoAcampamentoTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("GincanaNoAcampamento.php");

class GincanaNoAcampamentoTest extends TestCase {

    public function testGincanaNoAcampamento() {
        $obj = new GincanaNoAcampamento();
        //Test 1
        $want = "Fernanda";
        $got = $obj->winnerOfScavengerGame( [["Fernanda", "7"], ["Fernando", "9"], ["Gustavo", "11"]] );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
		//Test 2
		$want = "Pedro";
		$got = $obj->winnerOfScavengerGame( [["Maria", "7"], ["Pedro", "9"], ["Joao_Vitor", "5"], ["Isabel", "12"], ["Laura", "8"]] );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
		//Test 3
		$want = "Pedro";
		$got = $obj->winnerOfScavengerGame( [["Maria", "4"], ["Pedro", "3"], ["Joao", "2"]] );
        echo "\nTest 3: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }

}

File: GO/GincanaNoAcampamento_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import (
	"reflect"
	"testing"
)

func TestGincanaNoAcampamento(t *testing.T) {
	tables := []struct {
		inputs [][]string
		want   string
	}{
		{[][]string{{"Fernanda", "7"}, {"Fernando", "9"}, {"Gustavo", "11"}}, "Fernanda"},
		//Test 2
		{[][]string{{"Maria", "7"}, {"Pedro", "9"}, {"Joao_Vitor", "5"}, {"Isabel", "12"}, {"Laura", "8"}}, "Pedro"},
		//Test 3
		{[][]string{{"Maria", "4"}, {"Pedro", "3"}, {"Joao", "2"}}, "Pedro"},
	}

	for _, table := range tables {
		got := winnerOfScavengerGame(table.inputs)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %v but the return was this: %v", table.want, got)
		}
	}
}
